portal-bridge/src/bridge/era1.rs

```diff
@@ -79,50 +85,76 @@ impl Era1Bridge {
     }
 
     pub async fn launch(&self) {
-        info!("Launching era1 bridge");
+        info!("Launching era1 bridge: {:?}", self.mode);
+        match self.mode.clone() {
+            BridgeMode::FourFours(FourFoursModeType::Random) => self.launch_random().await,
+            BridgeMode::FourFours(FourFoursModeType::Epoch(epoch)) => {
+                self.launch_epoch(epoch).await
+            }
+            _ => panic!("4444s bridge only supports 4444s modes."),
+        }
+        info!("Bridge mode: {:?} complete.", self.mode);
+    }
+
+    pub async fn launch_random(&self) {
         for era1_path in self.era1_files.clone().into_iter() {
-            info!("Processing era1 file at path: {:?}", era1_path);
-            // We are using a semaphore to limit the amount of active gossip transfers to make sure
-            // we don't overwhelm the trin client
-            let gossip_send_semaphore = Arc::new(Semaphore::new(GOSSIP_LIMIT));
+            self.gossip_era1(era1_path).await;
+        }
+    }
 
-            let raw_era1 = self
-                .http_client
-                .get(era1_path.clone())
-                .recv_bytes()
-                .await
-                .unwrap_or_else(|_| panic!("unable to read era1 file at path: {era1_path:?}"));
-            let era1 = Era1::deserialize(&raw_era1).expect("to be able to deserialize era1 file");
-            let epoch_index = era1.block_tuples[0].header.header.number / EPOCH_SIZE as u64;
-            info!("Era1 file read successfully, gossiping block tuples for epoch: {epoch_index}");
-            let epoch_acc = match self.get_epoch_acc(epoch_index).await {
-                Ok(epoch_acc) => epoch_acc,
-                Err(e) => {
-                    error!("Failed to get epoch acc for epoch: {epoch_index}, error: {e}");
-                    continue;
-                }
-            };
-            let master_acc = Arc::new(self.header_oracle.master_acc.clone());
-            let mut serve_block_tuple_handles = vec![];
-            for block_tuple in era1.block_tuples {
-                let permit = gossip_send_semaphore
-                    .clone()
-                    .acquire_owned()
-                    .await
-                    .expect("to be able to acquire semaphore");
-                let serve_block_tuple_handle = Self::spawn_serve_block_tuple(
-                    self.portal_clients.clone(),
-                    block_tuple,
-                    epoch_acc.clone(),
-                    master_acc.clone(),
-                    permit,
-                );
-                serve_block_tuple_handles.push(serve_block_tuple_handle);
+    pub async fn launch_epoch(&self, epoch: u64) {
+        let epoch_string = epoch.to_string() + "-";
+        for era1_path in self.era1_files.clone().into_iter() {
```

Comment:
I think something like this might be a bit neater. It's odd to `return` in the middle of a function while the last line of a function is a `panic!`. Also, anytime we can avoid nesting, I tend to think it's worthwhile. Imo nested statements are typically confusing to read

```rust
let era1_path = self.era1_files.clone().into_iter().find(|file| file == &epoch_string);
match era1_path {
    Some(path) => self.gossip_era1(path).await,
    None => panic!("...."),
}
```